backend/src/main/java/com/woowacourse/friendogly/club/domain/Club.java

```diff
@@ -126,6 +137,78 @@ public static Club create(
                 .createdAt(LocalDateTime.now())
                 .imageUrl(imageUrl)
                 .build();
+        club.addClubMember(club.owner);
+        return club;
+    }
+
+    public void addClubMember(Member newMember) {
+        validateAlreadyExists(newMember);
+        validateMemberCapacity();
+        ClubMember clubMember = ClubMember.create(this, newMember);
+        clubMembers.add(clubMember);
+        clubMember.updateClub(this);
+    }
+
+
+    private void validateAlreadyExists(Member newMember) {
+        if (clubMembers.stream()
+                .anyMatch(clubMember -> Objects.equals(clubMember.getMember().getId(), newMember.getId()))
+        ) {
+            throw new FriendoglyException("이미 참여 중인 모임입니다.");
+        }
+    }
+
+    private void validateMemberCapacity() {
+        if (clubMembers.size() >= memberCapacity.getValue()) {
+            throw new FriendoglyException("최대 인원을 초과하여 모임에 참여할 수 없습니다.");
+        }
+    }
+
+    public void addClubPet(List<Pet> pets) {
+        pets.stream()
+                .peek(this::validateParticipatePet)
+                .map(pet -> new ClubPet(this, pet))
+                .peek(clubPet -> clubPet.updateClub(this))
+                .forEach(clubPets::add);
+    }
+
+    private void validateParticipatePet(Pet pet) {
+        if (!allowedGenders.contains(pet.getGender()) || !allowedSizes.contains(pet.getSizeType())) {
+            throw new FriendoglyException("모임에 데려갈 수 없는 강아지가 있습니다.");
+        }
+    }
+
+    public int countClubMember() {
+        return clubMembers.size();
     }
 
+    public boolean isEmpty() {
+        return clubMembers.isEmpty();
+    }
+
+    public void removeClubMember(Member member) {
+        ClubMember target = clubMembers.stream()
+                .filter(e -> e.getMember().getId().equals(member.getId()))
+                .findAny()
+                .orElseThrow(() -> new FriendoglyException("참여 중인 모임이 아닙니다."));
+        clubMembers.remove(target);
+        if (canDelegate(target)) {
+            this.owner = clubMembers.get(0).getMember();
+        }
+        target.updateClub(null);
+        removeClubPets(member);
+    }
+
+    private boolean canDelegate(ClubMember target) {
+        return owner.getId().equals(target.getMember().getId()) && !isEmpty();
+    }
```

Comment:
```suggestion
    private boolean canDelegate(Member member) {
        return owner.getId().equals(member.getId()) && !isEmpty();
    }
```

`removeClubMember()`의 파라미터랑 직접 비교하면
(+ 나중에 equals and hashCode 구현되면)
덜 복잡할 것 같은데 어떠신가요??

Reply:
entity에 동등성을 부여하는 부분은 논의해야 할 부분이라고 생각해서 따로 구현하진 않았어요.(너무 귀찮긴 했습니다.ㅎㅎ)